### ISSUE TITLE
Error reading avro file compressed with Snappy

### ISSUE THREAD

**Issue by boompig** [mtth/avsc#100]
I am running into issues when trying to read an avro file compressed with Snappy from python.

Code to read the Avro file:

```javascript
const avro = require('avsc');
const snappy = require("snappy");
var f5 = "foo-snappy.avro";
function readEvents(fname, useSnappy) {
    var events = [];
    console.log("Trying to read " + fname  + "...");
    console.time("avro-read-" + fname);
    var extra = null;
    if(useSnappy) {
        extra = {codecs: {"snappy": snappy.uncompress }};
        console.log("using snappy codec for fname " + fname);
    }
    avro.createFileDecoder(fname, extra)
        .on('metadata', (_, codec) => { console.log("codec = " + codec); })
        .on('data', (row) => {
            events.push(row);
        }).on("end", () => {
            console.log("done - read " + events.length + " rows");
            console.timeEnd("avro-read-" + fname);
        });
}
readEvents(f5, true);
```

Output:
```
Trying to read foo-snappy.avro...
using snappy codec for fname foo-snappy.avro
codec = snappy
events.js:163
      throw er; // Unhandled 'error' event
      ^

Error: Invalid input
```

Schema for the avro file:

```json
{
  "type" : "record",
  "namespace" : "com.foo.bar.baz",
  "name" : "Foo",
  "fields" :
	[
		{ "name" : "name", "type": ["null", "string"] },
		{ "name" : "val", "type": ["null", "int"] }
	]
}
```

Python code used to generate the avro file:

```python
import json
import fastavro as avro
events = [{"name": "Foo", "val": 3}]
def write_foo_avro(events, out_fname, codec):
    schema_file = "../filter-lambda/foo.avsc"
    with open(schema_file) as fp:
        schema = json.load(fp)
    with open(out_fname, "wb") as fp:
        avro.writer(fp, schema, events, codec=codec)
write_foo_avro(events, "../filter-lambda/foo-snappy.avro", "snappy")
```

node version: v7.8.0
Platform: MacOS Sierra 10.12.4
avsc version: "^5.0.1"

**Comment by boompig** [comment]
I found the root cause: the issue has been previously brought up in #61. However, while that issue is closed, the documentation has not been updated, and the "cryptic error" mentioned in the issue is no less cryptic.

I suggest at least:

1. Updating the documentation in the README to give instructions on how to use this module with snappy, including this workaround. Also helpful would be to add checksum validation code into the function referenced in #61.
2. Make the error resulting from this less cryptic.

However, I think you should reconsider adding Snappy support to this package, since Snappy is probably the most popular compression algo currently in use with Avro. I doubt I'm the only person who has run across this error and (almost) given up to use a different programming language with better Snappy+Avro library support.

**Comment by mtth** [comment]
Thanks for bringing this up. The documentation [had been updated](https://github.com/mtth/avsc/wiki/API#class-blockdecoderopts) but it's obviously not prominent enough. I'll change the file reading example in the README to use that codec (and if I have time later, adding the checksum handling in the API doc).

Regarding your other points:

+ I'll enhance change the error message to mention "decompression error". Any further clarifications (i.e. mentioning snappy) probably don't belong in `avsc`. Perhaps it'd be worth asking for features from these libraries (both more specific error messages and checksum support).
+ Snappy modules are usually heavy-weight (sometimes with C++ addons, so wouldn't play well with `browserify`) and there are several of them (so there probably isn't an optimal choice for all use-cases). Given how few lines of code it is to add it (once the checksum caveat is addressed), I still feel keeping it separate is the better decision.

**Comment by mtth** [comment]
Just released `5.0.2`with [more explicit error messages](https://github.com/mtth/avsc/commit/f6ecf1f29592212883b724eeeb939f1f1e064d5b), updated the README with a basic Snappy example, and added checksum validation to the [API's example](https://github.com/mtth/avsc/wiki/API#class-blockdecoderopts). Let me know if you have any other concerns.

**Comment by mtth** [comment]
Also added the codec to the error message (https://github.com/mtth/avsc/commit/bc1e46ced9c792cb7fe62713e1fb03f93af72945, to be included in the next release); feel free to reopen if you have any other concerns/suggestions.
